### ISSUE TITLE
Questions About Training, Stream ordering, and Labels

### ISSUE THREAD

**Issue by pclucas14** [RAIVNLab/InTheWild#2]
Hi, 

Thanks for open sourcing your code! I really enjoyed going through your paper. I have a few questions I was hoping you could answer. 

1) In the paper you mention that 
_For all experiments that require training, ..., we train each model for 4 epochs every 5,000 samples that are received_. 
If I understand correctly, this means methods like `Standard Training` and `Finetuning`  will stop and train 90k / 5k = 18 times during the sequential phase ? And other methods like `Prototype Finetuning` are updated on-the-fly without any offline training ? 
2) Could you give a little more info on how the samples in the stream are ordered ? Are the samples ordered in any way (e.g. all samples from class 1 are showed before moving on two class 2) or is it possible (although super unlikely but still) that the first 1K samples contain all 1K classes ? 
3) Finally, you mention in the `readme` that there exists a file called `overview.md` which lays out the code structure ? Could you link me to the file (as I could not find it) ?

Again, thanks for sharing your code!

**Comment by MattWallingford** [comment]
Hi,

Thanks for your interest in our paper and glad you enjoyed reading it!

1. Yes, you are correct that they train 90k/5k during the sequential phase. Prototype Tuning trains in the same fashion during the fine-tuning phase and on the fly only during the NCM phase. 

2. The sequence is randomly sampled from a heavy tailed distribution of all 1000 classes. Yes, all 1k classes could be drawn in the first 1k samples. The distribution being heavy tailed causes the class discovery to occur more slowly throughout the stream although it's true that the unseen classes appear much more frequently at the beginning. It's arguable that this is more natural since the more classes a system encounters in an environment, the less likely it is to encounter an additional unseen classes. 

3. Sorry about that, you're correct that the overview is missing. I will upload it this weekend. 

If you have any additional questions about the codebase, data, or paper feel free to ask.
-Matt

**Comment by pclucas14** [comment]
Hi Matt, 

thanks for the detailed answer! that clears things up. 

One more question : for the models that finetune the last layer, are the 250 classes overlapping with the initial Imagenet-1k such that the indices are shared ? Say during pretraining the class dog is at index 300, will it also be at 300 for the sequential training ? 

-Lucas

**Comment by MattWallingford** [comment]
Hey Lucas,

Yes, the indices are shared for all models. The pretrain imagenet-1k indices are in imagenet_classes.txt and the dictionary to map the 750 novel class to indices for a given sequence k are in SK/class_mapk.npy. The indices for the novel classes are reused across sequences so the labels will always go from 0-999 for each sequence. 

-Matt

**Comment by pclucas14** [comment]
Thanks! closing the issue
